kairon/shared/chat/user_media.py

```
@@ -53,28 +99,100 @@ def save_media_content(
-            logger.info(f"saved {media_id}")
+            url = CloudUtility.upload_file_bytes(binary_data, bucket, output_filename)
+            UserMedia.mark_user_media_data_upload_done(media_id=media_id,
+                                                       media_url=url,
+                                                       output_filename=output_filename,
+                                                       filesize=filesize)
+            logger.info(f"saved {media_id} successfully")
         except ClientError as e:
             logger.exception(e)
-            raise AppException("File upload failed")
+            UserMedia.mark_user_media_data_upload_failed(media_id=media_id, reason=str(e))
+            raise AppException(f"File upload for {media_id} failed")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)